File: iSangeet/app/src/main/java/com/example/isangeet/MainActivity.java
package com.example.isangeet;

import androidx.appcompat.app.AppCompatActivity;

import android.Manifest;
import android.os.Bundle;
import android.os.Environment;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

import com.karumi.dexter.Dexter;
import com.karumi.dexter.PermissionToken;
import com.karumi.dexter.listener.PermissionDeniedResponse;
import com.karumi.dexter.listener.PermissionGrantedResponse;
import com.karumi.dexter.listener.PermissionRequest;
import com.karumi.dexter.listener.single.PermissionListener;

import java.io.File;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        ListView listView;
        super.onCreate(savedInstanceState);
        listView = findViewById(R.id.listView);
        setContentView(R.layout.activity_main);
        Dexter.withContext(this)
                .withPermission(Manifest.permission.READ_EXTERNAL_STORAGE)
                .withListener(new PermissionListener() {
                    @Override
                    public void onPermissionGranted(PermissionGrantedResponse permissionGrantedResponse) {

                        Toast.makeText(MainActivity.this, "Runtime permission given", Toast.LENGTH_SHORT).show();
                        ArrayList<File> mySongs = fetchSongs(Environment.getExternalStorageDirectory());
                        String[] item =new String[mySongs.size()];
                        for (int i=0 ; i<mySongs.size() ; i++){
                            item[i] = mySongs.get(i).getName().replace("mp3" ,"");
                        }
                        ArrayAdapter<String> adapter = new ArrayAdapter<String>(MainActivity.this , android.R.layout.simple_list_item_1,item);
                        listView.setAdapter(adapter);
                    }

                    @Override
                    public void onPermissionDenied(PermissionDeniedResponse permissionDeniedResponse) {

                    }

                    @Override
                    public void onPermissionRationaleShouldBeShown(PermissionRequest permissionRequest, PermissionToken permissionToken) {
                        permissionToken.continuePermissionRequest();
                    }
                })
                .check();
    }
    public ArrayList<File> fetchSongs(File file){
       ArrayList arrayList = new ArrayList();
        File [] songs= file.listFiles();
        if (songs!=null){
            for (File myFile : songs){
                if (!myFile.isHidden() && myFile.isDirectory()){
                    arrayList.addAll(fetchSongs(myFile));
                }
                else {
                    if(myFile.getName().endsWith(".mp3") && !myFile.getName().startsWith(".")){
                        arrayList.add(myFile);
                    }
                }

            }
        }
        return arrayList;
    }

}

File: recycleview_guide/app/src/main/java/com/example/recycleview_guide/MainActivity.kt
package com.example.recycleview_guide

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.recycleview_guide.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private val binding by lazy { ActivityMainBinding.inflate(layoutInflater) }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setContentView(binding.root)

        val songsObject = mutableListOf<song>()
        songsObject.add(song("Hello" ,"Just the discription"))
        songsObject.add(song("how are you" ,"Just the discription"))
        songsObject.add(song("Hello" ,"Just the discription"))
        songsObject.add(song("Hello" ,"Just the discription"))
        songsObject.add(song("Hello" ,"Just the discription"))
        songsObject.add(song("Hello" ,"Just the discription"))
        songsObject.add(song("Hello" ,"Just the discription"))
        songsObject.add(song("Hello" ,"Just the discription"))
        songsObject.add(song("Hello" ,"Just the discription"))
        songsObject.add(song("Hello" ,"Just the discription"))
        songsObject.add(song("Hello" ,"Just the discription"))
        songsObject.add(song("Hello" ,"Just the discription"))
        songsObject.add(song("Hello" ,"Just the discription"))
        songsObject.add(song("Hello" ,"Just the discription"))
        binding.songlist.adapter = MyAdapter(songsObject)
        binding.songlist.layoutManager = LinearLayoutManager(this)

    }
}